core/iwasm/interpreter/wasm_interp_fast.c

```diff
@@ -5654,21 +5657,165 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
                 switch (opcode) {
                     /* Memory */
                     case SIMD_v128_load:
+                    {
+                        uint32 offset, addr;
+                        offset =
+                            read_uint32(frame_ip); // TODO Check with an offset!
+                        addr = GET_OPERAND(uint32, I32, 0);
+                        frame_ip += 2;
+                        addr_ret = GET_OFFSET();
+                        CHECK_MEMORY_OVERFLOW(16);
+                        PUT_V128_TO_ADDR(frame_lp + addr_ret, LOAD_V128(maddr));
+                        break;
+                    }
                     case SIMD_v128_load8x8_s:
+#if ENABLE_NEON == 1
```

Comment:
Should I do something similar to the invokeNative files where I define some function/symbol that is defined for each architecture in a separate file and put the opcode implementation there, or is it better inline here?

Reply:
my preference would be to not keep it here as it might grow a lot if we start supporting more architectures.